### ISSUE TITLE
face_colors doesn't color faces properly, but instead interpolates the colors

### ISSUE THREAD

**Issue by coderforlife** [mikedh/trimesh#1997]
If I draw a box with each triangle a solid color like so:

```python
import trimesh
import numpy as np
mesh = trimesh.primitives.Box(extents=(5, 5, 5))
mesh.visual.face_colors = np.array([
    [255, 0, 0, 255],
    [255, 128, 0, 255],
    [255, 255, 0, 255],
    [128, 255, 0, 255],
    [0, 255, 0, 255],
    [0, 255, 128, 255],
    [0, 255, 255, 255],
    [0, 128, 255, 255],
    [0, 0, 255, 255],
    [128, 0, 255, 255],
    [255, 0, 255, 255],
    [255, 0, 128, 255],
])
mesh.show()
```

There are no solid colors anywhere on the cube, everything is a gradient. When doing a face-visual (instead of a vertex-visual) you need to be using flat colors for each triangle. This may require a geometry shader to do properly so that each triangle can emit the same color to each of its vertices (and each vertex is emitted by one triangle instead of shared between triangles).

It is even more severe than just making gradients as entire faces can miss the designated color. For example, in the box, there is no spot that is red (to make it really obvious, keep the first color red and set all other colors to black - the entire cube is black with no hint of red anywhere).

I was using the trimesh library to highlight particular faces with particular computed properties, but several faces weren't showing up because of this issue.

**Comment by mikedh** [comment]
Hey, I think you might want `mesh.show(smooth=False)` which should probably be the default if `mesh.visual.kind == 'face'`,  which produces this:
<img width="665" alt="image" src="https://github.com/mikedh/trimesh/assets/1403074/6f01bf65-4a9b-49df-a91a-d19bc0cec186">

**Comment by coderforlife** [comment]
Exactly! Sorry I didn't find that option. Still a bit odd that with smoothing some face colors get completely dropped...

**Comment by coderforlife** [comment]
Reopening... `smooth=False` doesn't work in Jupyter notebook...

**Comment by paireks** [comment]
Yes, I have the same issue. Tried smooth=False and mesh.visual.kind == 'face', but it still keeps smoothing.
![2023-11-01_17h21_10](https://github.com/mikedh/trimesh/assets/47977819/dbf13e40-8cb1-4f34-8cb6-13dd52fad9cc)

**Comment by paireks** [comment]
Another interesting issue is the lack of transparency for faces:
![image](https://github.com/mikedh/trimesh/assets/47977819/052a1865-442f-4660-829f-c758ebb43be5)

You can check source code there: https://github.com/paireks/dotbimpy/blob/master/dotbimpy/other/DotbimToTrimeshScene.ipynb

3d file is there:
https://3dviewer.net/#model=https://raw.githubusercontent.com/paireks/dotbim/master/test/ExampleFiles/TestFilesFromArchicad/MulticolorHouse.bim

3d file from previous example is there:
https://3dviewer.net/#model=https://raw.githubusercontent.com/paireks/dotbim/master/test/ExampleFiles/TestFilesFromC%23/CubesWithFaceColorsAndWithout_WithFixedNormals.bim

**Comment by gtangg11** [comment]
Has this been resolved. It seems preventing interpolation in trimesh is just not possible.

ie this code still interpolates

```python
mesh = Trimesh(mesh.vertices, mesh.faces, smooth=False, process=False)
mesh.visual.vertex_colors = None
mesh.visual.face_colors = np.random.randint(0, 255, (len(mesh.faces), 3))
mesh.visual.kind == 'face'
return mesh
```

**Comment by gtangg11** [comment]
Ok, it seems the reason why face interpolation is happening is because openGL stores the data in vertex form as opposed to face form. Just run your mesh through this function, which preserves the ordering of faces but makes the vertices unique

```python
def duplicate_verts(mesh: Trimesh):
    """
    """
    verts = mesh.vertices[mesh.faces.reshape(-1), :]
    faces = np.arange(0, verts.shape[0])
    faces = faces.reshape(-1, 3)
    return Trimesh(vertices=verts, faces=faces, process=False)
```

Wish this would have been made clear before, which would have saved a lot of time for everyone (only occurred to me reading the src code of mesh.show()).
